epsagon/aws_sdk_v2_factories/sts_factory.go

```diff
@@ -0,0 +1,31 @@
+package epsagonawsv2factories
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/epsagon/epsagon-go/protocol"
+	"reflect"
+)
+
+// STSEventDataFactory to create epsagon Resource from aws.Request to STS
+func StsDataFactory(r *aws.Request, res *protocol.Resource, metadataOnly bool) {
+	handleSpecificOperations := map[string]specificOperationHandler{
+		"GetCallerIdentity": handleStsGetCallerIdentityRequest,
+	}
+	handleSpecificOperation(r, res, metadataOnly, handleSpecificOperations, nil)
+}
+
+func updateMetadataField(data reflect.Value, key string, res *protocol.Resource) {
+	value, ok := getFieldStringPtr(data, key)
+	if ok {
+		res.Metadata[key] = value
+	}
+}
+
+func handleStsGetCallerIdentityRequest(r *aws.Request, res *protocol.Resource, metadataOnly bool) {
+	outputValue := reflect.ValueOf(r.Data).Elem()
```

Comment:
Do we need this if metadataOnly is true?

Reply:
yes as this data shouldn't be sent if metadataOnly is true

Reply:
I think what @Alon-Katz meant is that you can add this line into the if !metadataOnly (as you do not need to execute it if metadataOnly is true)

Reply:
oh now I see. Right, will fix